packages/big-design/src/components/FileUploader/FileUploader.tsx

```diff
@@ -0,0 +1,229 @@
+import React, {
+  cloneElement,
+  isValidElement,
+  LabelHTMLAttributes,
+  useCallback,
+  useEffect,
+  useId,
+  useMemo,
+  useState,
+} from 'react';
+
+import { warning } from '../../utils';
+import { FormControlDescription, FormControlLabel } from '../Form';
+import { useInputErrors } from '../Form/useInputErrors';
+
+import { DropZone, DropZoneLocalization } from './DropZone';
+import { File } from './File';
+import { StyledFileUploaderWrapper } from './styled';
+import { getImagesPreview, validateFiles } from './utils';
+
+export interface FileUploaderLocalization extends DropZoneLocalization {
+  optional: string;
+}
+
+interface DropzoneConfig {
+  label?: string;
+  description?: string;
+}
+
+export interface FileValidationError {
+  file: File;
+  fileIdx: number;
+  message?: string;
+  type: string;
+}
+
+export interface ValidatorConfig {
+  message?: string;
+  type: string;
+  validator: (file: File) => boolean;
+}
+
+export const defaultLocalization: FileUploaderLocalization = {
+  optional: 'optional',
+  upload: 'Upload',
+};
+
+interface Props {
+  description?: React.ReactNode;
+  dropzoneConfig?: DropzoneConfig;
+  error?: React.ReactNode | React.ReactNode[];
+  files: File[];
+  isPreviewHidden?: boolean;
```

Comment:
Could you please add some context of why we need this property and what is a use case?

Reply:
We need this to hide the default view of uploaded files:
<img width="352" alt="Screenshot 2024-02-06 at 11 23 43" src="https://github.com/bigcommerce/big-design/assets/84462142/3dd4cff6-601b-4172-ab3a-c41acb64f831">

in order to make custom view if it is needed.

<img width="987" alt="Screenshot 2024-02-06 at 11 25 07" src="https://github.com/bigcommerce/big-design/assets/84462142/a1bb133e-1543-49db-a9f8-eb4f7b648321">